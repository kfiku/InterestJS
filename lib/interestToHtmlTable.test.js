/* eslint-env jest */
const Interest = require('../lib/interest.js')
const interestToHtmlTable = require('../lib/interestToHtmlTable.js')

describe('interestToHtmlTable without params', function () {
  it('should render html with interest without params', function () {
    const interest = new Interest(1000, 5, 5)
    const html = interestToHtmlTable(interest)
    expect(html).toMatchSnapshot()
  })

  it('should render html with interest with startAmount param', function () {
    const interest = new Interest(1000, 5, 5, { startAmount: 1000 })
    const html = interestToHtmlTable(interest)
    expect(html).toMatchSnapshot()
  })

  it('should render html with interest with tax param', function () {
    const interest = new Interest(1000, 5, 5, { tax: 19 })
    const html = interestToHtmlTable(interest)
    expect(html).toMatchSnapshot()
  })
})

describe('interestToHtmlTable with formatMoney param', function () {
  it('should render html with formatMoney and interest without params', function () {
    const interest = new Interest(1000, 5, 5)
    const formatMoney = (num) => num.toFixed(0)
    const html = interestToHtmlTable(interest, { formatMoney })
    expect(html).toMatchSnapshot()
  })
})

describe('interestToHtmlTable with translations param', function () {
  it('should render html with translations and interest with tax param', function () {
    const interest = new Interest(1000, 5, 5, { tax: 19 })
    const translations = {
      Capital: 'Kapita≈Ç',
      Interest: 'Odsetki',
      Tax: 'Podatek',
      Sum: 'Suma'
    }
    const html = interestToHtmlTable(interest, { translations })
    expect(html).toMatchSnapshot()
  })
})
