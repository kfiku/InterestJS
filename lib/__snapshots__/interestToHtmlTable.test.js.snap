// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interestToHtmlTable with formatMoney param should render html with formatMoney and interest without params 1`] = `
"<table>
<thead>
<tr>
<th></th>
<th>Capital</th>
<th>Interest</th>
<th>Sum</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>1000</td>
<td>4</td>
<td>1004</td>
</tr>
<tr>
<td>2</td>
<td>2000</td>
<td>8</td>
<td>2013</td>
</tr>
<tr>
<td>3</td>
<td>3000</td>
<td>13</td>
<td>3025</td>
</tr>
<tr>
<td>4</td>
<td>4000</td>
<td>17</td>
<td>4042</td>
</tr>
<tr>
<td>5</td>
<td>5000</td>
<td>21</td>
<td>5063</td>
</tr>
<tr>
<td>Sum</td>
<td>5000</td>
<td>63</td>
<td>5063</td>
</tr>
</tbody>
</table>
"
`;

exports[`interestToHtmlTable with translations param should render html with translations and interest with tax param 1`] = `
"<table>
<thead>
<tr>
<th></th>
<th>Kapita≈Ç</th>
<th>Odsetki</th>
<th>Podatek</th>
<th>Suma</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>1000.00</td>
<td>4.17</td>
<td>0.79</td>
<td>1003.38</td>
</tr>
<tr>
<td>2</td>
<td>2000.00</td>
<td>8.35</td>
<td>1.59</td>
<td>2010.14</td>
</tr>
<tr>
<td>3</td>
<td>3000.00</td>
<td>12.54</td>
<td>2.38</td>
<td>3020.30</td>
</tr>
<tr>
<td>4</td>
<td>4000.00</td>
<td>16.75</td>
<td>3.18</td>
<td>4033.87</td>
</tr>
<tr>
<td>5</td>
<td>5000.00</td>
<td>20.97</td>
<td>3.98</td>
<td>5050.86</td>
</tr>
<tr>
<td>Suma</td>
<td>5000.00</td>
<td>62.78</td>
<td>11.92</td>
<td>5050.86</td>
</tr>
</tbody>
</table>
"
`;

exports[`interestToHtmlTable without params should render html with interest with startAmount param 1`] = `
"<table>
<thead>
<tr>
<th></th>
<th>Capital</th>
<th>Interest</th>
<th>Sum</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>1000.00</td>
<td>8.33</td>
<td>2008.33</td>
</tr>
<tr>
<td>2</td>
<td>2000.00</td>
<td>12.53</td>
<td>3020.86</td>
</tr>
<tr>
<td>3</td>
<td>3000.00</td>
<td>16.75</td>
<td>4037.61</td>
</tr>
<tr>
<td>4</td>
<td>4000.00</td>
<td>20.99</td>
<td>5058.60</td>
</tr>
<tr>
<td>5</td>
<td>5000.00</td>
<td>25.24</td>
<td>6083.84</td>
</tr>
<tr>
<td>Sum</td>
<td>5000.00</td>
<td>83.84</td>
<td>6083.84</td>
</tr>
</tbody>
</table>
"
`;

exports[`interestToHtmlTable without params should render html with interest with tax param 1`] = `
"<table>
<thead>
<tr>
<th></th>
<th>Capital</th>
<th>Interest</th>
<th>Tax</th>
<th>Sum</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>1000.00</td>
<td>4.17</td>
<td>0.79</td>
<td>1003.38</td>
</tr>
<tr>
<td>2</td>
<td>2000.00</td>
<td>8.35</td>
<td>1.59</td>
<td>2010.14</td>
</tr>
<tr>
<td>3</td>
<td>3000.00</td>
<td>12.54</td>
<td>2.38</td>
<td>3020.30</td>
</tr>
<tr>
<td>4</td>
<td>4000.00</td>
<td>16.75</td>
<td>3.18</td>
<td>4033.87</td>
</tr>
<tr>
<td>5</td>
<td>5000.00</td>
<td>20.97</td>
<td>3.98</td>
<td>5050.86</td>
</tr>
<tr>
<td>Sum</td>
<td>5000.00</td>
<td>62.78</td>
<td>11.92</td>
<td>5050.86</td>
</tr>
</tbody>
</table>
"
`;

exports[`interestToHtmlTable without params should render html with interest without params 1`] = `
"<table>
<thead>
<tr>
<th></th>
<th>Capital</th>
<th>Interest</th>
<th>Sum</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>1000.00</td>
<td>4.17</td>
<td>1004.17</td>
</tr>
<tr>
<td>2</td>
<td>2000.00</td>
<td>8.35</td>
<td>2012.52</td>
</tr>
<tr>
<td>3</td>
<td>3000.00</td>
<td>12.55</td>
<td>3025.07</td>
</tr>
<tr>
<td>4</td>
<td>4000.00</td>
<td>16.77</td>
<td>4041.84</td>
</tr>
<tr>
<td>5</td>
<td>5000.00</td>
<td>21.01</td>
<td>5062.85</td>
</tr>
<tr>
<td>Sum</td>
<td>5000.00</td>
<td>62.85</td>
<td>5062.85</td>
</tr>
</tbody>
</table>
"
`;
